src/lineup/internal/LineUpPanelActions.ts

```diff
@@ -217,78 +220,46 @@ export default class LineUpPanelActions extends EventHandler {
-      </ul>
-    `;
-
-    // Listen for row selection and update number of selected rows
-    // Show/hide some dropdown menu points accordingly using CSS
-    this.provider.on(LocalDataProvider.EVENT_SELECTION_CHANGED + '.download-menu', (indices: number[]) => {
-      (<HTMLElement>node.querySelector('[data-num-selected-rows]')).dataset.numSelectedRows = indices.length.toString();
-    });
-
-    const links = Array.from(node.querySelectorAll('a'));
-    for (const link of links) {
-      link.onclick = (evt) => {
-        evt.preventDefault();
-        evt.stopPropagation();
-        const type = link.dataset.t;
-        const onlySelected = link.dataset.s === 's';
-        exportLogic(<any>type, onlySelected, this.provider).then(({content, mimeType, name}) => {
-          this.downloadFile(content, mimeType, name);
-        });
-      };
-    }
-
-
-    return node;
-  }
-
-  private appendSaveRanking() {
-    const listener = (ranking: Ranking) => {
-      this.saveRankingDialog(ranking.getOrder());
-    };
-
-    return this.createMarkup(i18n.t('tdp:core.lineup.LineupPanelActions.saveEntities'), 'fa fa-save', listener);
-  }
-
-  private appendExtraButtons() {
+  private appendExtraButtons(parent: HTMLElement) {
     const buttons = <IRankingButtonExtensionDesc[]>listPlugins(EXTENSION_POINT_TDP_RANKING_BUTTON);
     return buttons.map((button) => {
       const listener = () => {
         button.load().then((p) => this.scoreColumnDialog(p));
       };
-      return this.createMarkup(button.title, 'fa ' + button.cssClass, listener);
+
+      const luButton = new PanelRankingButton(parent, this.provider, button.title, 'fa ' + button.cssClass, listener);
+      this.header.addButton(luButton);
     });
   }
 
-  private downloadFile(content: BufferSource | Blob | string, mimeType: string, name: string) {
-    const doc = this.node.ownerDocument;
-    const downloadLink = doc.createElement('a');
-    const blob = new Blob([content], {type: mimeType});
-    downloadLink.href = URL.createObjectURL(blob);
-    (<any>downloadLink).download = name;
+  private appendExtraTabs() {
+    const plugins = <IPanelTabExtensionDesc[]>listPlugins(EP_TDP_CORE_LINEUP_PANEL_TAB).sort((a, b) => a.order - b.order);
+    plugins.forEach((plugin) => {
+      let isLoaded = false;
+      const tab = new PanelTab(this.tabContainer.node, plugin);
 
-    doc.body.appendChild(downloadLink);
-    downloadLink.click();
-    downloadLink.remove();
-  }
+      const onClick = () => {
+        if (isLoaded) {
+          if (this.collapse) {
+            this.collapse = false; // expand side panel
+          }
+          this.tabContainer.showTab(tab);
 
-  protected saveRankingDialog(order: number[]) {
-    editDialog(null, (name, description, sec) => {
-      this.fire(LineUpPanelActions.EVENT_SAVE_NAMED_SET, order, name, description, sec);
+        } else {
+          plugin.load().then((p) => {
```

Comment:
The interface for `factory` is `factory(parent: HTMLElement, provider: LocalDataProvider, desc: IRankingButtonExtensionDesc, idType: IDType, extraArgs: object): Promise<IScoreParam>;` according to `IPanelTabExtension`

so when you add the type `IPanelTabExtension` to p then the method is incompatible to the interface's definition. for example `tab.events` is passed as IDType

Reply:
Thank you, fixing in next commit.

Reply:
If you could take a look again at the commit to confirm it looks better now.